kore/src/Kore/Variables/Fresh.hs

```diff
@@ -58,6 +49,43 @@ instance FreshVariable Variable where
         pivotMin = variable { variableCounter = Nothing }
         fixSort var = var { variableSort = variableSort variable }
 
+{- | Rename one set of variables while avoiding another.
+
+If any of the variables to rename occurs in the set of avoided variables, it
+will be mapped to a fresh name in the result. Every fresh name in the result
+will also be unique among the fresh names.
+
+To use @refreshVariables@ with 'Kore.Step.Pattern.substitute', map the result
+with 'Kore.AST.Valid.mkVar':
+
+@
+'Kore.Step.Pattern.substitute' ('Kore.AST.Valid.mkVar' \<$\> refreshVariables avoid rename) :: 'Kore.Step.Pattern.StepPattern' Object Variable -> 'Kore.Step.Pattern.StepPattern' Object Variable
```

Comment:
Is there a nice way of fitting in 80 chars? If not, wouldn't this look better with a line split before "::"?